### PR TITLE
SCIPROD-1585: [CLIBCO] error for non-and logic on primary entity and field

### PR DESCRIPTION
**Pull request by csuloway-dnanexus**
Raise exception and exit with error when a cohort filter has primary entity and field logic that is not "and".

### CODE REVIEW

File: src/python/dxpy/cli/dataset_utilities.py [hunk 1]
```diff
@@ -1212,7 +1212,10 @@ def create_cohort(args):
     entity = rec_descriptor.model["global_primary_key"]["entity"]
     field = rec_descriptor.model["global_primary_key"]["field"]
     filters = resp.get("filters", {})
-    payload = cohort_final_payload(samples, entity, field, filters, from_project)
+    try:
+        payload = cohort_final_payload(samples, entity, field, filters, from_project)
+    except Exception as e:
+        err_exit("{}: {}".format(entity_result["id"], e))
```
Comment (piader1) [inline src/python/dxpy/cli/dataset_utilities.py:1218]: Does `format()` transform type `Exception` to `str` by default?

Comment (csuloway-dnanexus) [reply src/python/dxpy/cli/dataset_utilities.py:1218]: Good catch. `str.format` should use the string value of the `Exception`
> ...Returns a copy of the string where each replacement field is replaced with the string value of the corresponding argument.
https://docs.python.org/3/library/stdtypes.html?highlight=str%20format#str.format
(also tested for Python 2 & 3)

File: src/python/dxpy/dx_extract_utils/cohort_filter_payload.py [hunk 2]
```diff
@@ -20,55 +20,57 @@ def generate_pheno_filter(values, entity, field, filters):
         # entity field values cannot be selected by adding a filter to the
         # existing compound pheno_filter. Move the existing pheno_filter into a
         # new pheno_filter level compound filter.
-        print("WARNING: pheno_filter compound filter logic is not \"and\", creating a nested compound filter."
-              " The Cohort Browser may not work as expected with this cohort.")
+        raise ValueError("Invalid input cohort. Cohorts must have “and” logic on the primary entity and field.")
         filters["pheno_filters"] = {"compound": [filters["pheno_filters"]], "logic": "and"}
```
Comment (Niranjanpandeshwar) [inline src/python/dxpy/dx_extract_utils/cohort_filter_payload.py:24]:
```suggestion
```

File: src/python/dxpy/dx_extract_utils/cohort_filter_payload.py [hunk 3]
```diff
@@ -20,55 +20,57 @@ def generate_pheno_filter(values, entity, field, filters):
         # entity field values cannot be selected by adding a filter to the
         # existing compound pheno_filter. Move the existing pheno_filter into a
```
Comment (Niranjanpandeshwar) [inline src/python/dxpy/dx_extract_utils/cohort_filter_payload.py:21]: I guess this comment needs to be updated to reflect new changes.

**Pull request merged**